src/screens/GroupInfo/hooks/useGroupInfo.js

```diff
@@ -317,14 +318,18 @@ const useGroupInfo = (channelId = null) => {
 
   const blockAnonUser = async () => {
     try {
+      const blockedUsername = selectedUser?.user?.anon_user_info_color_name
+        ? getOfficialAnonUsername(selectedUser?.user)
+        : selectedUser?.user?.anonymousUsername || selectedUser?.user?.username;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The changes made in the `blockAnonUser` function are good as they now use the `getOfficialAnonUsername` function to get the official anonymous username for blocking.
```diff
- const blockedUsername = selectedUser?.user?.anonymousUsername || selectedUser?.user?.username;
+ const blockedUsername = selectedUser?.user?.anon_user_info_color_name
+   ? getOfficialAnonUsername(selectedUser?.user)
+   : selectedUser?.user?.anonymousUsername || selectedUser?.user?.username;
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->